unreleased_history/performance_improvements/java_api_consistency.md

```diff
@@ -0,0 +1,16 @@
+* Java API extensions to improve consistency and completeness of APIs
+  1 Extended `RocksDB.get( ... ByteBuffer key, ByteBuffer value ...)` which now accepts indirect buffer parameters as well as direct buffer parameters
+  2 Extended `RocksDB.put( ... ByteBuffer key, ByteBuffer value ...)` which now accepts indirect buffer parameters as well as direct buffer parameters
+  3 Added `RocksDB.merge( ... ByteBuffer key, ByteBuffer value ...)` methods with the same parameter options as `put(...)` - direct and indirect buffers are supported
+  4 Added `RocksIterator.key( ... byte[] key, byte[] value ...)` methods which retrieve the iterator key into the supplied buffer
```

Comment:
These two API signatures seem to be slightly off.

Reply:
Ah yes, they do. I've tidied up all the signatures a bit..